tensorflow/core/common_runtime/sycl/sycl_util.h

```diff
@@ -21,17 +21,58 @@ limitations under the License.
 #define TENSORFLOW_CORE_COMMON_RUNTIME_SYCL_SYCL_UTIL_H_
 
 #include "tensorflow/core/common_runtime/device.h"
+#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
 // For DMA helper
 #include "tensorflow/core/common_runtime/dma_helper.h"
 #include "tensorflow/core/framework/tensor.h"
 
 namespace tensorflow {
-  inline void* GetBase(const Tensor* src) {
-    return const_cast<void*>(DMAHelper::base(src));
-  }
+inline void const* GetBase(const Tensor* src) { return DMAHelper::base(src); }
+inline void* GetBase(Tensor* dst) { return DMAHelper::base(dst); }
 
-  inline void* GetBase(Tensor* dst) { return DMAHelper::base(dst); }
+inline void SYCLmemcpy(Eigen::SyclDevice const& device,
+                       Tensor const& src_tensor, Tensor* dst_tensor) {
+  size_t size = src_tensor.TotalBytes();
```

Comment:
I will fix `size` to be const in the next commit ( waiting on the testing to finish ) 
the `src_ptr` is already const 
in order to make `dst_ptr` const we would have to change the interfance of device.memcpy ( it won't compile with const at present )